static/app/components/dropdownMenuV2.tsx

```diff
@@ -0,0 +1,301 @@
+       * state.selectionManager.isSelected([key]) = true
+       */
+      return isSection
+        ? [...node.childNodes].some(child =>
+            state.selectionManager.isSelected(`${child.key}`)
+          )
+        : state.selectionManager.isSelected(`${node.key}`);
+    });
+    setHasFocus(isLeafSubmenu);
+  }, [state.selectionManager.selectedKeys]);
+  /**
+   * Menu props from useMenu, modified to disable keyboard events
+   * if the current menu does not have focus.
+   */
+  const modifiedMenuProps = {
+    ...menuProps,
+    ...(!hasFocus && {
+      onKeyUp: () => null,
+      onKeyDown: () => null,
+    }),
+  };
+
+  /**
+   * Render a single menu item
+   */
+  const renderItem = (node: Node<MenuItemProps>, isLastNode: boolean) => {
+    return (
+      <MenuItem
+        node={node}
+        isLastNode={isLastNode}
+        state={state}
+        onClose={closeRootMenu}
+        closeOnSelect={closeOnSelect}
+      />
+    );
+  };
+
+  /**
+   * Render a submenu whose trigger button is a menu item
+   */
+  const renderItemWithSubmenu = (node: Node<MenuItemProps>, isLastNode: boolean) => {
+    const trigger = ({props: submenuTriggerProps, ref: submenuTriggerRef}) => (
+      <MenuItem
+        ref={submenuTriggerRef}
+        renderAs="div"
+        node={node}
+        isLastNode={isLastNode}
+        state={state}
+        isSubmenuTrigger
+        {...submenuTriggerProps}
+      />
+    );
+
+    return (
+      <MenuControl
+        items={node.value.children as MenuItemProps[]}
+        trigger={trigger}
+        menuTitle={node.value.submenuTitle}
+        placement="right top"
+        offset={-4}
+        crossOffset={-8}
+        closeOnSelect={closeOnSelect}
+        isOpen={state.selectionManager.isSelected(node.key)}
+        isSubmenu
+        closeRootMenu={closeRootMenu}
+        closeCurrentSubmenu={() => state.selectionManager.clearSelection()}
+        renderWrapAs="li"
+      />
+    );
+  };
+
+  /**
+   * Render a collection of menu items
+   */
+  const renderCollection = (collection: Node<MenuItemProps>[]) =>
+    collection.map((node, i) => {
+      const isLastNode = collection.length - 1 === i;
+      const showSeparator =
+        !isLastNode && (node.type === 'section' || collection[i + 1]?.type === 'section');
+      const {separatorProps} = useSeparator({elementType: 'li'});
+
+      let itemToRender: ReactNode;
+
+      if (node.type === 'section') {
+        itemToRender = (
+          <MenuSection node={node}>{renderCollection([...node.childNodes])}</MenuSection>
+        );
+      } else {
+        itemToRender = node.value.isSubmenu
+          ? renderItemWithSubmenu(node, isLastNode)
+          : renderItem(node, isLastNode);
+      }
+
+      return (
+        <Fragment key={node.key}>
+          {itemToRender}
+          {showSeparator && <Separator {...separatorProps} />}
+        </Fragment>
+      );
+    });
+
+  return (
+    <FocusScope restoreFocus autoFocus>
```

Comment:
Ahh, we should consider replacing this https://github.com/getsentry/sentry/blob/9ffb055577cce118fb51042273bc57b90b73329f/static/app/components/globalModal/index.tsx#L6 with this.

Though I can imagine there's probably some great modal stuff in this aria library

Reply:
(as in FocusScope looks good, let's get rid of focus-trap?)

Reply:
yeah I'd want to use [useDialog](https://react-spectrum.adobe.com/react-aria/useDialog.html#example) once I get to modals, so we can probably just replace `focus-trap` then